python-backend/tests/variances/resources/test_variance_resource.py

```diff
@@ -68,3 +69,93 @@ def test_get_variances_application_status_filter_list(self, test_client, db_sess
         assert all(map(
             lambda v: v['variance_application_status_code'] in status_codes,
             get_data['records']))
+
+    def test_get_variances_application_filter_by_mine_region(self, test_client, db_session, auth_headers):
+        """Should filter variances by mine region"""
+        region_code = "NW"
+        mineWith_region_NW = MineFactory(mine_region='NW')
+        mineWith_region_SW = MineFactory(mine_region="SW")
+        batch_size = 3
+        VarianceFactory.create_batch(size=batch_size)
+        VarianceFactory(mine=mineWith_region_NW)
+        VarianceFactory(mine=mineWith_region_SW)
+
+        get_resp = test_client.get(f'/variances?region={region_code}',headers=auth_headers['full_auth_header'])
+        get_data = json.loads(get_resp.data.decode())
+        print(get_data)
+        assert get_resp.status_code == 200
+        assert all(map(
+            lambda v: v['mine_name'] == mineWith_region_NW.mine_name,
+            get_data['records']))
+
+    def test_get_variances_application_filter_by_major_mine(self, test_client, db_session, auth_headers):
+        """Should filter variances by major vs regional mine"""
+        major_mine = MineFactory(major_mine_ind=True)
+        regional_mine = MineFactory(major_mine_ind=False)
+
+        VarianceFactory(mine=major_mine)
+        VarianceFactory(mine=regional_mine)
+        get_resp = test_client.get(f'/variances?major=f', headers=auth_headers['full_auth_header'])
+        get_data = json.loads(get_resp.data.decode())
+        assert get_resp.status_code == 200
+        assert all(map(
+            lambda v: v['mine_name'] == regional_mine.mine_name,
+            get_data['records']))
+
+    def test_get_variances_application_filter_by_compliance_code(self, test_client, db_session, auth_headers):
+        """Should filter variances by compliance code"""
+        compliance_codes = [54,66]
```

Comment:
You can just generate 2 random compliance codes, which is better than hardcoding them

Reply:
done